codeworld-api/src/CodeWorld/Driver.hs

```diff
@@ -1089,7 +1095,7 @@ getMousePos canvas = do
 onEvents :: Element -> (Event -> IO ()) -> IO ()
 onEvents canvas handler = do
     Just window <- currentWindow
-    on window keyDown $ do
+    _ <- on window keyDown $ do
```

Comment:
I thought it was interesting that the discarded result has type `IO ()`. Smells funny to me. But maybe it's fine.

```
src/CodeWorld/Driver.hs:1098:5: error: [-Wunused-do-bind, -Werror=unused-do-bind]
    A do-notation statement discarded a result of type ‘IO ()’
    Suppress this warning by saying
      ‘_ <- on window keyDown
              $ do code <- getKeyCode =<< event
                   let keyName = keyCodeToText code
                   when (keyName /= "")
                     $ do liftIO $ handler (KeyPress keyName)
                          preventDefault
                          ....
                   key <- getKey =<< event
                   ....’
     |
1098 |     on window keyDown $ do
     |     ^^^^^^^^^^^^^^^^^^^^^^...
```

Reply:
This is expected.  `on` registers the listener, and returns an action to unregister it.  We don't care about the unregister action.